libtransmission/peer-msgs.cc

```diff
@@ -913,6 +913,12 @@ void tr_peerMsgsImpl::parse_ltep(MessageReader& payload)
     {
         parse_ltep_handshake(payload);
 
+        // The peer most likely supports LTEP, so send our LTEP handshake in
```

Comment:
I guess I'm not following your logic.  If you think we should always send the LTEP handshake (one of the two things I don't understand), then just always send it, no reason to way until we are reading back a handshake, do it unilaterally in the peerMsgs creator. I think I am missing something more subtle that you are seeing.

Reply:
> If you think we should always send the LTEP handshake

That's not what I'm thinking. Does this table make it clearer? Transmission won't send the LTEP handshake in cases not listed below.

| Peer set LTEP flag | Peer sent LTEP handshake | Result |
| :--: | :--: | :-- |
| Y | N/A | Transmission sends LTEP handshake on establishing the connection |
| N | Y | Transmission replies with LTEP handshake after parsing incoming LTEP handshake |

Reply:
Ah, I understand your intent now.  

So, further, "Peer set LTEP flag=N, Peer sent LTEP handshake=N : TR does not (ever) send_ltep_handshake" ?

I'll note again that there is no client in the wild that sends LTEP handshake without setting the flag, but I'm fine with whatever you decide, glad we made this explicit.

Reply:
If you think this makes sense, please give me a go ahead, as I want to move the changes from your PR to a new PR. I feel like there's 2 topics in 1 PR now.

> So, further, "Peer set LTEP flag=N, Peer sent LTEP handshake=N : TR does not (ever) send_ltep_handshake" ?

Yes, exactly.

Reply:
> If you think the code now makes sense, please give me a go ahead, as I want to move the changes from your PR to a new PR. I feel like there's 2 topics in 1 PR now.
> 
> > So, further, "Peer set LTEP flag=N, Peer sent LTEP handshake=N : TR does not (ever) send_ltep_handshake" ?
> 
> Yes, exactly.

I cannot seem to clear the "Unresolved conversations" flag in github?  I hid/resolved each comments for it still shows as Unresolved overall.